treeherder/intermittents_commenter/commenter.py

```diff
@@ -0,0 +1,285 @@
+        self.dry_run = kwargs['dry_run']
+        self.session = self.new_request()
+
+    def run(self):
+        startday, endday = self.calculate_date_strings(self.weekly_mode, 6)
+        alt_startday, alt_endday = self.calculate_date_strings(True, DISABLE_DAYS)
+        all_params = self.create_comments(startday, endday, alt_startday, alt_endday)
+        self.print_or_submit_comments(all_params)
+
+    def create_comments(self, startday, endday, alt_startday, alt_endday):
+        """Posts a bug comment containing stats to each bug whose total number of
+           occurrences (daily or weekly) meet the appropriate threshold."""
+
+        bug_stats = self.get_bug_stats(startday, endday)
+        alt_bug_stats = self.get_bug_stats(alt_startday, alt_endday)
+        test_run_count = self.get_test_runs(startday, endday)
+
+        all_params = []
+        template = Template(self.open_file('comment.template', False))
+
+        if self.weekly_mode:
+            top_bugs = [bug[0] for bug in sorted(bug_stats.items(), key=lambda x: x[1]['total'],
+                        reverse=True)][:TOP_BUGS_THRESHOLD]
+
+        for bug_id, counts in bug_stats.iteritems():
+            change_priority = None
+            bug_info = None
+            whiteboard = None
+            priority = 0
+            rank = None
+
+            # recommend disabling when more than 150 failures tracked over 21 days
+            if alt_bug_stats[bug_id]['total'] >= DISABLE_THRESHOLD:
+                bug_info, whiteboard = self.check_bug_info(bug_info, bug_id)
+
+                if not self.check_whiteboard_status(whiteboard):
+                    priority = 3
+                    whiteboard = self.update_whiteboard(whiteboard, WHITEBOARD_DISABLE_RECOMMENDED)
+
+            if self.weekly_mode:
+                priority = self.assign_priority(priority, counts)
+                if priority == 2:
+                    bug_info, whiteboard = self.check_bug_info(bug_info, bug_id)
+                    change_priority, whiteboard = self.check_needswork_owner(change_priority, bug_info, whiteboard)
+
+                if (counts['total'] < UNKNOWN_THRESHOLD):
+                    bug_info, whiteboard = self.check_bug_info(bug_info, bug_id)
+                    whiteboard = self.check_needswork(whiteboard)
+
+                if bug_id in top_bugs:
+                    rank = top_bugs.index(bug_id)+1
+            else:
+                bug_info, whiteboard = self.check_bug_info(bug_info, bug_id)
+                change_priority, whiteboard = self.check_needswork_owner(change_priority, bug_info, whiteboard)
+
+            text = template.render(bug_id=bug_id,
+                                   total=counts['total'],
+                                   test_run_count=test_run_count,
+                                   rank=rank,
+                                   priority=priority,
+                                   failure_rate=round(counts['total']/float(test_run_count), 3),
```

Comment:
Could you add `from __future__ import division` to this file, to ensure this division is Python 3 compatible. (Bug 1452420  did this for the rest of the codebase; see also https://eev.ee/blog/2016/07/31/python-faq-how-do-i-port-to-python-3/#division-always-produces-a-float)